### PR TITLE
Added handling of conversion of names 

### PR DESCRIPTION
**Pull request by jonwarghed**
when given a csv with incorrect column names

closes issue #14 

https://github.com/bxparks/bigquery-schema-generator/issues/33

### CODE REVIEW

File: bigquery_schema_generator/generate_schema.py [hunk 1]
```diff
@@ -111,6 +112,13 @@ def __init__(self,
         self.line_number = 0
         self.error_logs = []
 
+        # This option generally wants to be turned on as any inferred schema
+        # will not be accepted by `bq load` when it contains illegal characters.
+        # Characters such as #, / or -. Neither will it be accepted if the column name
+        # in the schema is larger than 128 characters. 
+        self.sanitize_names = sanitize_names
```
Comment (bxparks) [inline bigquery_schema_generator/generate_schema.py]: Can you remove the trailing white space here?

File: bigquery_schema_generator/generate_schema.py [hunk 2]
```diff
@@ -111,6 +112,13 @@ def __init__(self,
...
+
```
Comment (bxparks) [inline bigquery_schema_generator/generate_schema.py]: Here too, remove trailing white spaces..

File: bigquery_schema_generator/generate_schema.py [hunk 3]
```diff
@@ -670,14 +680,17 @@ def flatten_schema_map(schema_map,
...
+                                                   sorted_schema, sanitize_names)
             elif key == 'type' and value in ['QINTEGER', 'QFLOAT', 'QBOOLEAN']:
                 new_value = value[1:]
             elif key == 'mode':
                 if infer_mode and value == 'NULLABLE' and filled:
                     new_value = 'REQUIRED'
                 else:
                     new_value = value
+            elif key == 'name' and sanitize_names:
+                print(value)
```
Comment (bxparks) [inline bigquery_schema_generator/generate_schema.py]: Remove debugging print() statement, will corrupt the schema output

File: tests/test_generate_schema.py [hunk 4]
```diff
@@ -428,19 +428,21 @@ def verify_data_chunk(self, chunk_count, chunk):
...
         quoted_values_are_strings = ('quoted_values_are_strings' in data_flags)
+        sanitize_names = ('sanitize_names' in data_flags)
         records = chunk['records']
         expected_errors = chunk['errors']
         expected_error_map = chunk['error_map']
         expected_schema = chunk['schema']
 
         print("Test chunk %s: First record: %s" % (chunk_count, records[0]))
-
+        print(sanitize_names)
```
Comment (bxparks) [inline tests/test_generate_schema.py]: Remove debugging print() statement

**Comment by jonwarghed** [comment]
Hi again. It should look better now, let me know if there is anything else you think I should change.

**Comment by bxparks** [comment]
There is still one line that has a trailing space, but it's minor. I'll fix it after merging this, and I'll update the CHANGELOG and README.

Thanks for the PR!

**Comment by jonwarghed** [comment]
Thank you, I really like the work you put in.

**Pull request merged**